Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportRenderer.cpp

```diff
@@ -224,6 +224,22 @@ namespace EMStudio
         return m_entityContext->GetContextId();
     }
 
+    void AnimViewportRenderer::CheckBounds()
+    {
+        AZ::Vector3 groundPos;
+        AZ::TransformBus::EventResult(groundPos, m_groundEntity->GetId(), &AZ::TransformBus::Events::GetWorldTranslation);
+
+        // Used the projected position to check the distance.
+        AZ::Vector3 characterProjectedPos = GetCharacterCenter();
+        characterProjectedPos.SetZ(groundPos.GetZ());
+
+        float distance = groundPos.GetDistanceEstimate(characterProjectedPos);
+        if (distance > BoundMaxDistance)
+        {
+            AZ::TransformBus::Event(m_groundEntity->GetId(), &AZ::TransformBus::Events::SetWorldTranslation, characterProjectedPos);
+        }
+    }
```

Comment:
Doesn't this result in pops when it actually moves the ground?

What I would do is something like this:

```
const float tileSize = X; // We need to find out this value.
const float newX = std::fmod(charPos.x, tileSize);
const float newY = std::fmod(charPos.y, tileSize);
groundPlane.SetPos(newX, newY, oldZ);
```

fmod works like modulo on integers and returns the remainder of the division. If we use fmod with the tile size, so every tile on the checkerboard. We basically move the checkerboard along with the character and just displace it a little bit within range [0, tileSize]. And we just do that every frame.